x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/convert/UrlDecodeTests.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.expression.function.scalar.convert;
+
+import com.carrotsearch.randomizedtesting.annotations.Name;
+import com.carrotsearch.randomizedtesting.annotations.ParametersFactory;
+
+import org.elasticsearch.xpack.esql.core.expression.Expression;
+import org.elasticsearch.xpack.esql.core.tree.Source;
+import org.elasticsearch.xpack.esql.core.type.DataType;
+import org.elasticsearch.xpack.esql.expression.function.AbstractUrlEncodeDecodeTestCase;
+import org.elasticsearch.xpack.esql.expression.function.FunctionName;
+import org.elasticsearch.xpack.esql.expression.function.TestCaseSupplier;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+
+@FunctionName("url_decode")
+public class UrlDecodeTests extends AbstractUrlEncodeDecodeTestCase {
+
+    public UrlDecodeTests(@Name("TestCase") Supplier<TestCaseSupplier.TestCase> testCaseSupplier) {
+        this.testCase = testCaseSupplier.get();
+    }
+
+    @ParametersFactory
+    public static Iterable<Object[]> parameters() {
+        List<TestCaseSupplier> suppliers = new ArrayList<>();
+        String evaluatorToString = "UrlDecodeEvaluator[val=Attribute[channel=0]]";
+
+        for (DataType dataType : DataType.stringTypes()) {
+            suppliers.add(new TestCaseSupplier(List.of(dataType), () -> createTestCaseWithRandomUrl(dataType, evaluatorToString, false)));
+
+            for (TestCaseSupplier.TypedDataSupplier supplier : TestCaseSupplier.stringCases(dataType)) {
+                TestCaseSupplier testCaseSupplier = new TestCaseSupplier(
+                    supplier.name(),
+                    List.of(supplier.type()),
+                    () -> createTestCaseWithRandomString(dataType, evaluatorToString, false, supplier)
+                );
```

Comment:
Nit: For readability, you can probable make `createTestCaseWithRandomString()` return the full `TestCaseSupplier`. Same for `createTestCaseWithRandomUrl()`

Reply:
If this is what I think this is: in the parameters() method we can't call any "random...()" method, as seed wasn't initialized. They have to be delayed to the `test()` method calls.
This means that all the code currently inside `createTestCaseWithRandomUrl(...)` would still have to be inside the lambda:
```
                TestCaseSupplier testCaseSupplier = new TestCaseSupplier(
                    supplier.name(),
                    List.of(supplier.type()),
                    () -> // Here
                );
```

Actually, I believe that's the reason the "TestCaseSupplier" exists to begin with: To allow for that later call to random methods.

The change you made looks nice to me. If you want to try and make this change too, up to you. Thanks!